rust/src/traits/cast/mod.rs

```diff
@@ -483,6 +483,30 @@ impl RoundCast<bool> for String {
     }
 }
 
+impl<R: Round> RoundCast<f32> for FBig<R> {
+    fn round_cast(v: f32) -> Fallible<Self> {
+        FBig::try_from(v).map_err(|_| err!(FailedCast, "value must be finite"))
+    }
+}
+
+impl<R: Round> RoundCast<f64> for FBig<R> {
+    fn round_cast(v: f64) -> Fallible<Self> {
+        FBig::try_from(v).map_err(|_| err!(FailedCast, "value must be finite"))
+    }
+}
+
+impl<R: Round> RoundCast<FBig<R>> for f32 {
```

Comment:
Oh: I think this answers my earlier question: I didn't realize you could implement traits for builtins.

Reply:
Yes, there's something called the orphan rules in Rust:
https://smallcultfollowing.com/babysteps/blog/2015/01/14/little-orphan-impls/

The key requirement being that you cannot implement external traits for external types. In this case, `RoundCast` is an internal trait, so the impl is allowed by the compiler. Vice-versa, for example I can implement `From<ErrorTypeFromAnotherCrate>` (a foreign trait) for `opendp::error::Error` (an internal type) to convert errors from another crate into errors from the opendp crate.